openwisp_controller/config/static/config/js/default_templates.js

```diff
@@ -46,23 +47,81 @@ django.jQuery(function ($) {
                 });
             });
         },
-        bindDefaultTemplateLoading = function (urls) {
+        getRelevantTemplates = function (urls) {
+            var orgID = $('#id_organization').val(),
+                backend = $('#id_config-0-backend').val();
+            if (!orgID || !backend) {
+                return;
+            }
+            // proceed only if an organization and a backend have been selected
+            if (orgID.length === 0 || backend.length === 0) {
+                return;
+            }
+
+            var url = urls[orgID];
+            // get relevant templates of selected org and backend
+            url = url + '?backend=' + backend;
+            $.get(url).done(function (data) {
+                var relevantTemplates = {};
+                $.each(data.templates, function (i, uuid) {
+                    relevantTemplates[uuid] = uuid;
+                });
+                var disabledTemplates =
+                    localStorage.getItem('config-disabled-templates', null);
+                disabledTemplates = JSON.parse(disabledTemplates) || {};
+                $('input.sortedm2m').each(function () {
+                    if (($(this).val() in relevantTemplates) === false) {
+                        // hide templates which are not relevant
+                        // also removed disabled attribute and uncheck it
+                        if ($(this).prop('disabled')) {
+                            disabledTemplates[$(this).val()] = $(this).val();
+                            $(this).prop('disabled', false);
+                            $(this).prop('checked', false);
+                        }
+                        $(this).parent().parent().hide();
+                    } else {
+                        // show templates which are relevant
+                        if ($(this).val() in disabledTemplates) {
+                            $(this).prop('disabled', true);
+                            $(this).prop('checked', true);
+                            delete disabledTemplates[$(this).val()];
+                        }
+                        $(this).parent().parent().show();
+                    }
+                });
+                disabledTemplates = JSON.stringify(disabledTemplates);
+                localStorage.setItem('config-disabled-templates', disabledTemplates);
```

Comment:
Can't we just keep it in the memory instead of using localStorage? How making `disabledTemplates` persistent helpful?